maa-cli/src/copilot.rs

```diff
@@ -0,0 +1,175 @@
+use anyhow::Context;
+use reqwest::blocking::Client;
+use serde_json::Value;
+use std::{
+    fs::{self, File},
+    io::{BufReader, Write},
+};
+
+use crate::{dirs::Dirs, installer::maa_core::find_resource};
+
+pub fn copilot(dirs: &Dirs, link: String, disable_formation: bool, loop_times: Option<u8>) -> bool {
+    let api = "https://prts.maa.plus/copilot/get/";
+    let tasks_dir = format!("{}/tasks/", dirs.config().display());
+    let base_resource_dir = find_resource(dirs).context("Failed to find resource!");
+    let stage_dir = format!(
+        "{}/Arknights-Tile-Pos",
+        base_resource_dir.unwrap().to_string_lossy()
+    );
+    let copilot_id = match link.get(link.len() - 5..) {
+        Some(id) => id,
+        None => {
+            println!("Invaild maa code.");
+            return false;
+        }
+    };
+    let url = format!("{}{}", api, copilot_id);
+    let mut task_type: &str = "Copilot";
+
+    // Visit the API to obtain the data of homework.
+    let client = Client::new();
+    let response = client.get(&url).send().unwrap();
+    let json: Value = response.json().unwrap();
+    let status_code = json["status_code"].clone().to_string();
+    if status_code != "200" {
+        println!("Request failed");
+        return false;
+    }
+
+    // Extract the response
+    let context = json["data"]["content"].clone();
+    let content = match context.as_str() {
+        Some(content) => content.to_string(),
+        None => {
+            println!("Error parsing JSON");
+            return false;
+        }
+    };
+
+    let recast: Value = serde_json::from_str(&content).unwrap();
+
+    let content_type = recast["type"].clone();
+    if content_type == "SSS" {
+        task_type = "SSSCopilot";
+    }
+
+    // Print stage info
+    println!("Stage info:\n");
+    if task_type == "Copilot" {
+        let stage_code_name = recast["stage_name"].as_str().unwrap();
+        let stage_name = find_stage_name(stage_code_name, stage_dir);
+        println!("{}\n", stage_name);
+    } else {
+        let stage_name = recast["stage_name"].as_str();
+        println!("{}\n", stage_name.unwrap());
+    }
+
+    //Print operators info
+    println!("Operator lists:\n");
+    let opers = recast["opers"].clone();
+    for operator in opers.as_array().unwrap() {
+        let name = operator["name"].as_str().unwrap();
+        let skill = operator["skill"].as_u64().unwrap();
+        println!("{}, Skill: {};", name, skill,);
+    }
+
+    // Save json file
+    let json_file_name = format!(".{}.json", copilot_id);
+    let json_path = format!("{}{}", tasks_dir, json_file_name);
+    write_file(content, json_path.clone());
+
+    // Write a task file for maa-cli for this assignment
+    let loop_times_not_null: u8 = if let Some(value) = loop_times {
```

Comment:
I suggest that the `loop_times` and `formation` parameters can be inputs instead of command-line options.